src/main/scala/za/co/absa/hyperdrive/trigger/api/rest/services/HdfsService.scala

```diff
@@ -0,0 +1,171 @@
+/*
+ * Copyright 2018 ABSA Group Limited
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package za.co.absa.hyperdrive.trigger.api.rest.services
+
+import com.fasterxml.jackson.databind.ObjectMapper
+import com.fasterxml.jackson.module.scala.DefaultScalaModule
+import org.apache.commons.io.IOUtils
+import org.apache.hadoop.fs.{FileSystem, Path, PathFilter}
+import org.apache.spark.deploy.SparkHadoopUtil
+import org.slf4j.LoggerFactory
+import org.springframework.stereotype.Service
+
+import java.nio.charset.StandardCharsets.UTF_8
+import javax.inject.Inject
+import scala.io.Source
+
+trait HdfsService {
```

Comment:
This is not just HdfsService, it can also parse file and  it understands spark structured streaming.

Reply:
loginUserFromKeytab should not be a public interface and it should be used only in this class.
Also methods parseKafkaOffsetStream and getLatestOffsetFilePath should be somewhere else, if you want to have a hdfs service, it should contain only read file, list dirs, list files and so on

Reply:
Yeah need to rethink this. Tbh, the other methods are only public for simplified testing